openverse_catalog/dags/common/loader/reporting.py

```diff
@@ -39,10 +47,16 @@ def report_completion(provider_name, duration, record_counts_by_media_type):
         duration = humanize_time_duration(duration)
 
     # List record count per media type
-    media_type_reports = "\n".join(
-        f"  - `{media_type}`: {record_count or '_No data_'}"
-        for media_type, record_count in record_counts_by_media_type.items()
-    )
+    media_type_reports = ""
+    for media_type, (loaded, upserted) in record_counts_by_media_type.items():
+        media_type_reports += f"  - `{media_type}`: {upserted or '_No data_'}"
+        if upserted is None or loaded is None:
+            # Can't make calculation without data
+            continue
+        duplicates = loaded - upserted
+        if duplicates:
+            media_type_reports += f" _({duplicates} duplicates)_"
```

Comment:
I'm having a hard time following the logic across all the files, but is `duplicates` coming from the `affected_count` in `_clean_intermediate_table_data`? If so, then labeling them "duplicates" is incorrect, as the `_clean_intermediate_table_data` function also removes rows missing certain required columns (as well as removing duplicates).

If we want the duplicate count specifically, do we need to partition the affected counting against the missing required rows and the duplicates separately?

Reply:
We get the `duplicates` by subtracting the number of records that got `upserted` from the number that were `loaded`. We retrieve those counts from XCom; they're pushed [here](https://github.com/WordPress/openverse-catalog/blob/1ace8ea118c9360ec1e0d3469f0988c182b5637c/openverse_catalog/dags/common/loader/loader.py#L51). So the `loaded_count` is the count that comes back from `load_s3_data_to_intermediate_table`, which already [takes into account](https://github.com/WordPress/openverse-catalog/blob/1ace8ea118c9360ec1e0d3469f0988c182b5637c/openverse_catalog/dags/common/loader/sql.py#L160) rows that get cleaned and doesn't count them. So I think it should be good!

@AetherUnbound was testing changes to actually report the count of cleaned rows in addition to duplicates though -- did those changes get pushed or is that going to be a separate PR?

Reply:
Oh my god, I forgot to push :sob: Doing that now, so sorry!

Reply:
I can add some more documentation to the reporting function here to better describe what we're seeing though!